text/0003-data-frames.md

```diff
@@ -0,0 +1,252 @@
+- Feature Name: data_frames
+- Start Date: 2020-08-05
+- RFC PR: [nushell/rfcs#0003](https://github.com/nushell/rfcs/pull/3)
+- Nushell Issue: [nushell/nushell#0000](https://github.com/nushell/nushell/issues/0000)
+
+# Summary
+
+[summary]: #summary
+
+This RFC merges the Row and Table Value types into a single new value type: Frame. Data frames take inspiration from data processing systems like R and Pandas. Data frames will play the fundamental role of modelling data in Nu and will have enough descriptive power to describe all forms of structure, including streaming tables, lists, and objects.
+
+# Motivation
+
+[motivation]: #motivation
+
+The current system has a few unexpected limitations:
+
+- The top-level rows represent a table of rows, but it's unclear how to represent a top-level list of strings vs a stream of strings.
+- A similar ambiguity exists between an "object" (a data structure denoted by key/value pairs) and a table of one row
+- Inner-tables are modelled differently than top-level tables, leading to confusion
+- There is no way to currently represent a matrix
+- As rows are streamed instead of tables, it's not possible to predict how to display this information. This is currently mitigated by buffering some number of rows and treating them as one "table"
+- Likewise, since rows are streamed instead of tables, it's unclear what the user should expect if they request a column that is not present, as this column may appear in the following row instead.
+- When table data is serialized, there is a large amount of duplication, as columns are repeated with each row sent.
+- Additionally, there is currently no way to represent rows using a row literal. We propose a frame literal that will fill this role.
+
+# Guide-level explanation
+
+[guide-level-explanation]: #guide-level-explanation
+
+Data frame representation:
+
+```rust
+struct DataFrame {
```

Comment:
Since I'm not familiar with nu's current representations, repeating that here for a quick comparison could help.